coreos-installer

```diff
@@ -613,9 +613,23 @@ log() {
 #########################################################
 write_image_to_disk() {
     log "Writing disk image"
-    
+
+    set -o pipefail
+    # Check for gzip and xz correspondingly
+    if [ "$(dd count=2 bs=1 if=/mnt/dl/imagefile.gz)" = "$(printf '\x1f\x8b')" ]
+    then
+        DECOMPRESSION_TOOL='zcat'
+    elif [ "$(dd count=6 bs=1 if=/mnt/dl/imagefile.gz)" = "$(printf '\xfd\x37\x7a\x58\x5a\x00')" ]
+    then
+        DECOMPRESSION_TOOL='xzcat'
+    else
+        log "Compressed file format is not supported."
```

Comment:
How about also mentioning in that error also the formats that *are* supported?

Reply:
Sure, thanks for pointing that out!